functions/module_metadata/assert.pp

```diff
@@ -0,0 +1,69 @@
+# Fails a compile if the client system is not compatible with the module's
+# `metadata.json`
+#
+# @param module_name
+#   The name of the module that should be checked
+#
+# @param options
+#   @see $simplib::module_metadata::blacklist::options
+#
+# @return [None]
+#
+function simplib::module_metadata::assert (
+  String[1] $module_name,
+  Optional[Struct[{
+    enable => Optional[Boolean],
+    blacklist => Optional[Array[Variant[String[1], Hash[String[1], Variant[String[1], Array[String[1]]]]]]],
+    blacklist_validation => Optional[Struct[{
+      enable => Optional[Boolean],
```

Comment:
Why not just not specify the structure if you don't want to blacklist?  Seems confusing to specify the structure but turn off the capability.

Reply:
Because Hiera stacking overrides. Taking something out of a deep_merged hash is pretty much impossible. Turning something off is easy.

Reply:
So you think the parameters called into the assert functions will be exposed by the class/define calling the assert function so that it can be overwritten in Hiera?    Sounds like you want the user to be able to override the check.  In other words you want the user to be able to say 'trust me, I know better  what you support' ?

Reply:
Why bother with the asserts if you allow the user to ignore?

Reply:
Instead, why not expose a single variable to apply metadata.json checks or not and wrap the assert function call in if $check_metadata.  If you are dealing with a power user who wants to turn it off, keep it simple.  The power user assumes the full burden.

Reply:
@lnemsick-simp This isn't for power users, this is for people that are throwing up their hands because things should work and don't want to read our code. One global switch to kill it all (just like we have now) and minor switches if you really know what you're doing (usually devs).